catalogs/usfimr/build_catalog.py

```diff
@@ -0,0 +1,125 @@
+    parser.add_argument('--shapefile', required=True)
+    args = parser.parse_args()
+
+    shp_path = args.shapefile
+
+    print('Targeting shapefile at path {}'.format(shp_path))
+
+    items = []
+    running_spatial_extent = (0, 0, 0, 0)
+    root_path = "./catalog"
+    running_start_dt = None
+    running_end_dt = None
+    with fiona.open(shp_path) as fc:
+        for feature in fc:
+            geom = feature['geometry']
+            props = feature['properties']
+
+            if props['Start_Time'] is not None:
+                start_time = time.fromisoformat(props['Start_Time'])
+            else:
+                start_time = datetime.min.time()
+
+            if props['End_Time'] is not None:
+                end_time = time.fromisoformat(props['End_Time'])
+            else:
+                end_time = datetime.min.time()
+
+
+            start_dt = datetime.combine(date.fromisoformat(props['Flood_Date']), start_time)
+            end_dt = datetime.combine(date.fromisoformat(props['Flood_Date']), end_time)
+            shapely_geom = shape(geom)
+            bbox = shapely_geom.bounds
+            bbox_list = [bbox[0], bbox[1], bbox[2], bbox[3]]
+
+            running_spatial_extent = (min(running_spatial_extent[0], bbox[0]), max(running_spatial_extent[1], bbox[1]), min(running_spatial_extent[2], bbox[2]), max(running_spatial_extent[3], bbox[3]))
+
+            if running_start_dt:
+                if running_start_dt > start_dt:
+                    running_start_dt = start_dt
+            else:
+                running_start_dt = start_dt
+
+            if running_end_dt:
+                if running_end_dt < end_dt:
+                    running_end_dt = end_dt
+            else:
+                running_end_dt = end_dt
+
+            fid = feature['id']
+            
+            binary_asset = Asset(
+                href="./{}-usfimr.wkb".format(fid),
+                description="well known binary representation",
+                media_type="application/wkb"
+            )
+            text_asset = Asset(
+                href="./{}-usfimr.wkt".format(fid),
+                description="well known text representation",
+                media_type="application/wkt"
+            )
+            json_asset = Asset(
+                href="./{}-usfimr.geojson".format(fid),
+                description="geojson representation",
+                media_type="application/geo+json"
+            )
+            serializable_convex_hull = mapping(shapely_geom.convex_hull)
+            item = Item(fid, serializable_convex_hull, bbox_list, start_dt, deepcopy(props))
+            item.add_asset(key="wkt", asset=text_asset)
+            item.add_asset(key="wkb", asset=binary_asset)
+            item.add_asset(key="geojson", asset=json_asset)
+            items.append(item)
+
+            os.makedirs('{}/{}'.format(root_path, fid), exist_ok=True)
+
+            with open('{}/{}/{}-usfimr.wkt'.format(root_path, fid, fid), 'w') as wkt_file:
```

Comment:
Doesn't really matter here since we're always on unix like systems, but consider `os.path.join` for constructing file paths.